include/ruby/thread_native.h

```diff
@@ -201,5 +201,20 @@ void rb_native_cond_initialize(rb_nativethread_cond_t *cond);
  */
 void rb_native_cond_destroy(rb_nativethread_cond_t *cond);
 
+typedef struct rb_thread_hook_event_args {
```

Comment:
rb_thread_hook_event_args and rb_thread_hook should not be exposed.

Reply:
I think it's necessary to expose `rb_thread_hook_event_args` as it's what contains the `waiting` value.

E.g. an inexpensive way to measure thread contention is to look at how many threads are waiting when a thread is paused or resumed.

For `rb_thread_hook` I'll look at how to make it an opaque type.

Reply:
Ah I understand that `waiting` is how many threads are chained in ready queue.

If we expose the value, we need to maintain the `waiting` counter in future... but maybe no problem. ... maybe...
This counter is depends on current implementation. For example, if we introduce queues per priority, we need to sum all of threads per queues.

Or writing a comment that the definition is implementation detailed (not guaranteed).

But again, it will not be a big issue in future... maybe...

Reply:
I misunderstand `waiting` counter is based on current master, but you added.
IMO only for this purpose I don't want to introduce new expensive counter.
At least, they should not use atomic operations.

Reply:
I removed the `waiting` counter.